Source/Charts/Charts/BarLineChartViewBase.swift

```diff
@@ -533,7 +533,7 @@ open class BarLineChartViewBase: ChartViewBase, BarLineScatterCandleBubbleChartD
             
             let h = getHighlightByTouchPoint(recognizer.location(in: self))
             
-            if h === nil || h!.isEqual(self.lastHighlighted)
+            if h === nil || h?.isEqual(self.lastHighlighted) ?? false
```

Comment:
this can be just `if h?.isEqual(self.lastHighlighted) ?? true {`

Reply:
The former force unwrapping was also acceptable, because if the first condition passed on the or statement, we would have already entered the if statement.

Reply:
done